server/api/product.js

```diff
@@ -82,8 +84,14 @@ router.put('/review/:reviewId', async (req, res, next) => {
 // DELETE api/product/:productId
 router.delete('/:productId', async (req, res, next) => {
   try {
-    const {params: {productId}} = req
-    await Product.destroy({where: {id: productId}})
+    if (!req.user) throw new Error('Not logged in')
+    const {user: {merchantId}, params: {productId}} = req
+    await (await Product.findByPk(productId)).setMerchant(1)
+    const destroyed = await Product.destroy({
+      where: {id: productId, merchantId},
+      returning: true
+    })
+    if (!destroyed) throw new Error('You are not the merchant of this item')
```

Comment:
no need to fix, just a comment:
when you provide non existing productId, you might still get **Error('You are not the merchant of this item')**